component/otelcol/processor/tail_sampling/types.go

```diff
@@ -197,6 +207,87 @@ func (spanCountCfg SpanCountCfg) Convert() tsp.SpanCountCfg {
 	return otelCfg
 }
 
+// BooleanAttributeCfg holds the configurable settings to create a boolean attribute filter
+// sampling policy evaluator.
+type BooleanAttributeCfg struct {
+	// Tag that the filter is going to be matching against.
+	Key string `river:"key,attr"`
+	// Value indicate the bool value, either true or false to use when matching against attribute values.
+	// BooleanAttribute Policy will apply exact value match on Value
+	Value bool `river:"value,attr"`
+}
+
+func (booleanAttributeCfg BooleanAttributeCfg) Convert() tsp.BooleanAttributeCfg {
+	var otelCfg tsp.BooleanAttributeCfg
+
+	mustDecodeMapStructure(map[string]interface{}{
+		"key":   booleanAttributeCfg.Key,
+		"value": booleanAttributeCfg.Value,
+	}, &otelCfg)
+
+	return otelCfg
+}
+
+// The error mode determines whether to ignore or propagate
+// errors with evaluating OTTL conditions.
+type ErrorMode string
+
+const (
+	// "ignore" causes evaluation to continue to the next statement.
+	ErrorModeIgnore ErrorMode = "ignore"
+	// "propagate" causes the evaluation to be false and an error is returned.
+	ErrorModePropagate ErrorMode = "propagate"
+)
+
+var (
+	_ river.Validator          = (*ErrorMode)(nil)
+	_ encoding.TextUnmarshaler = (*ErrorMode)(nil)
+)
+
+// Validate implements river.Validator.
+func (e *ErrorMode) Validate() error {
+	var ottlError ottl.ErrorMode
+	return ottlError.UnmarshalText([]byte(string(*e)))
+}
+
+// Convert the River type to the Otel type
+func (e *ErrorMode) Convert() ottl.ErrorMode {
+	var ottlError ottl.ErrorMode
+	ottlError.UnmarshalText([]byte(string(*e)))
+	return ottlError
+}
+
+func (e *ErrorMode) UnmarshalText(text []byte) error {
+	str := ErrorMode(strings.ToLower(string(text)))
+	switch str {
+	case ErrorModeIgnore, ErrorModePropagate:
+		*e = str
+		return nil
+	default:
+		return fmt.Errorf("unknown error mode %v", str)
+	}
+}
+
+// OttlConditionCfg holds the configurable setting to create a OTTL condition filter
+// sampling policy evaluator.
+type OttlConditionCfg struct {
```

Comment:
The name here feels misaligned with our normal naming convention; this would usually be `OTTLConditionConfig`.

Reply:
Ok, I changed the name of this type and all the other types in this file (they all used to end with `Cfg`)